### PR TITLE
Introduce cloud run|deploy commands 

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
Introduce `cloud run|deploy` commands.
Designed to work as the conventional `run|deploy` commands but build in the cloud instead of locally.

**NB!** Remove latest commit before merging

Ping @rosen-vladimirov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "mobile-cli-lib": "https://github.com/telerik/mobile-cli-lib/tarball/master",
     "mocha": "3.5.3",
     "mocha-typescript": "1.1.8",
-    "nativescript": "https://github.com/NativeScript/nativescript-cli/tarball/master",
+    "nativescript": "https://github.com/NativeScript/nativescript-cli/tarball/kerezov/cloud-deploy-run",
```
Comment (Fatme) [inline package.json]: Not needed :smile:

File: lib/commands/cloud-run.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+			getOutputDirectory: this.$nsCloudBuildService.getServerOperationOutputDirectory.bind(this.$nsCloudBuildService),
+			buildPlatform: this.$nsCloudBuildCommandHelper.buildPlatform.bind(this.$nsCloudBuildCommandHelper)
+		});
+	}
+
+	public async canExecute(args: string[]): Promise<boolean> {
+		await this.$nsCloudEulaCommandHelper.ensureEulaIsAccepted();
+
+		if (args.length) {
+			this.$errors.fail("This input is not valid for the run command");
```
Comment (rosen-vladimirov) [inline lib/commands/cloud-run.ts]: for the `cloud run` command maybe

File: lib/commands/cloud-deploy.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	public async canExecute(args: string[]): Promise<boolean> {
+		this.$bundleValidatorHelper.validate();
+		await this.$nsCloudEulaCommandHelper.ensureEulaIsAccepted();
+
+		if (!args || !args.length) {
+			this.$errors.fail("Provide platform.");
+		}
+
+		if (args.length > 1) {
+			this.$errors.fail("Only a single platform is supported.");
```
[the remark this answers is not in the document]
Comment (Mitko-Kerezov) [reply lib/commands/cloud-deploy.ts]: This is the logic in the original `tns deploy` command - it does not support both blatforms

File: lib/commands/cloud-deploy.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+		return this.$platformService.deployPlatform(deployPlatformInfo);
+	}
+
+	public async canExecute(args: string[]): Promise<boolean> {
+		this.$bundleValidatorHelper.validate();
+		await this.$nsCloudEulaCommandHelper.ensureEulaIsAccepted();
+
+		if (!args || !args.length) {
+			this.$errors.fail("Provide platform.");
```
[the remark this answers is not in the document]
Comment (Mitko-Kerezov) [reply lib/commands/cloud-deploy.ts]: As stated earlier - `tns deploy` does not support both platforms and I thought neither should `tns cloud deploy` then

File: lib/commands/cloud-build.ts [hunk 5]
```diff
@@ -11,7 +11,8 @@ export class CloudBuild implements ICommand {
 		private $nsCloudBuildService: ICloudBuildService,
 		private $nsCloudOptionsProvider: ICloudOptionsProvider,
 		private $options: ICloudOptions,
-		private $projectData: IProjectData) {
+		private $projectData: IProjectData,
+		private $bundleValidatorHelper: IBundleValidatorHelper) {
```
[the remark this answers is not in the document]
Comment (Mitko-Kerezov) [reply lib/commands/cloud-build.ts]: Fixed with latest commit
@rosen-vladimirov

File: lib/commands/cloud-run.ts (hunk 2 above)
Comment (Mitko-Kerezov) [reply lib/commands/cloud-run.ts]: Fixed

File: package.json (hunk 1 above)
Comment (Mitko-Kerezov) [reply package.json]: @Fatme it's needed for successful transpilation - I've marked the commit with `[delete]`

**Pull request merged**